tests/ui/pages/desktop/base.py

```diff
@@ -26,6 +31,16 @@ def login(self, email, password):
     def logout(self):
         self.header.click_logout()
 
+    def wait_for_page_to_load(self):
+        if 'PYTEST_BASE_URL' in os.environ:
```

Comment:
Logic inside page objects is a red flag, especially without comments explaining it. A base URL can be provided in many ways, not just via environment variable. You can also use a base URL for a local instance, so I suspect this logic is invalid.